### PR TITLE
Firestore setup

### PR DESCRIPTION
**Pull request by gwynethsc**
Adds firestore config files and establishes realtime connection between application and Firestore via the project service.

### CODE REVIEW

File: angular-app/src/app/app.component.ts [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent implements AfterViewInit {
+export class AppComponent implements OnInit, AfterViewInit {
+
   @ViewChild('chat') public chat: MatSidenav;
+  public user$;
```
Comment (Google-JM) [inline angular-app/src/app/app.component.ts:15]: My recollection of private vs public in angular is kinda fuzzy. Why does user$ need to be public?

File: angular-app/src/app/app.component.ts [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
 
-  constructor(private toggleChatService: ToggleChatService) {
+  constructor(
+    public authService: AuthService,
```
Comment (Google-JM) [inline angular-app/src/app/app.component.ts]: Similar question, why does authService need to be public?

File: angular-app/src/app/auth.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { TestBed } from '@angular/core/testing';
+
+import { AuthService } from './auth.service';
+
+describe('AuthService', () => {
```
Comment (Google-JM) [inline angular-app/src/app/auth.service.spec.ts:5]: Is the build setup to run these tests automatically already?

**Review by Google-JM — APPROVED** [review]
Looking great! Couple of small questions on some of this setup.

**Review by leah-mayo — APPROVED** [review]

File: angular-app/src/app/app.component.ts (hunk 1 above)
Comment (gwynethsc) [reply angular-app/src/app/app.component.ts:15]: Private properties are unfortunately not visible to my template either; I use this in the ngIf directive in the HTML, which is why it's public.

File: angular-app/src/app/app.component.ts (hunk 2 above)
Comment (gwynethsc) [reply angular-app/src/app/app.component.ts]: Oops! This one is actually fine as private; I forgot to change it when I was messing around with other things trying to get it to work. Thanks for the catch!

File: angular-app/src/app/auth.service.spec.ts (hunk 3 above)
Comment (gwynethsc) [reply angular-app/src/app/auth.service.spec.ts:5]: No, they're run with ng test using some external checker, but I haven't been very good about the whole test-driven development process. Thanks for the reminder, especially now that we're moving more into non-UI elements.

**Pull request merged**